index.d.ts

```diff
@@ -0,0 +1,59 @@
+declare namespace DbTypes {
+    type ClientType = 'mysql' | 'sqlite' | 'postgres';
+
+    interface DbErrorOpts {
+        nativeError: {
```

Comment:
This is an instance of `Error`. Do we even need to expose the `DBError` constructor since these errors shouldn't be created by the user?

Reply:
- `DbErrorOpts` is not exported, it's used internally in `DBError` constructor.
- `DBError` is exported from your module (https://github.com/Vincit/db-errors/blob/master/lib/dbErrors.js#L52), in types it's exported to correspond with existing code

Reply:
I did find a somewhat legitimate use case for this.

Was testing a postgresql error and found an exception not yet implement (PR incoming 😄 )

```
    query.onError((err) => {
      // Workaround until this is implemented in db errors
      // @ts-ignore
      if (err.code === '23P01') {
        return Promise.reject(
          // @ts-ignore
          new CheckViolationError({
            // @ts-ignore
            table: err.table,
            // @ts-ignore
            constraint: err.constraint,
            // @ts-ignore
            nativeError: err,
          })
        );
      }

      return Promise.reject(wrapError(err));
    });
```

So this is a bit easier than pushing a fork but typescript does not enjoy it.